src/os/bluestore/PMEMDevice.cc

```diff
@@ -217,9 +217,7 @@ int PMEMDevice::write(uint64_t off, bufferlist& bl, bool buffered)
 {
   uint64_t len = bl.length();
   dout(20) << __func__ << " " << off << "~" << len  << dendl;
-  assert(len > 0);
-  assert(off < size);
-  assert(off + len <= size);
+  assert(is_valid_io(off, len));
```

Comment:
Yes, PMEMDevice is based on byte address.  So offset && length are not limited by block_size (in fact by directIO requirement)